### ISSUE TITLE
Inconsistent line spacing in switch cases

### ISSUE THREAD

**Issue by kamilogorek** [prettier/prettier#814]
Input:

```js
switch (foo) {
  case "baz":
    doThing();
  case "bar":
    doThing();

  case "baz":
    doThing();


  case "qux":
    doThing();

  default:
    doThing();
}
```

Expected output:

```js
switch (foo) {
  case "baz":
    doThing();

  case "bar":
    doThing();

  case "baz":
    doThing();

  case "qux":
    doThing();

  default:
    doThing();
}
```

Actual output:
```js
switch (foo) {
  case "baz":
    doThing();
  case "bar":
    doThing();

  case "baz":
    doThing();

  case "qux":
    doThing();

  default:
    doThing();
}
```

Related https://github.com/prettier/prettier/issues/805

**Comment by vjeux** [comment]
We've tried at the beginning to enforce those kind of new lines but it didn't really work. Instead what prettier does is to keep \n where the users put them, but will collapse all the series on newlines to just a single one.

**Comment by kamilogorek** [comment]
Sounds very reasonable, however, it's contradictory to main readme statement :)

> ...completely ignoring any of the original formatting. Say hello to completely consistent syntax!

**Comment by yamafaktory** [comment]
I may have something regarding that one, will create a PR when it's done.

**Comment by vjeux** [comment]
That sentence is the aspirational goal of the project. We've had to make a few concessions in order to make the project usable in practice.

**Comment by jlongster** [comment]
Yeah, the readme needs to be updated. I haven't changed it since releasing.

**Comment by jlongster** [comment]
I'm not sure if we want to enforce spaces here, as someone may do:

```
case foo: return 1;
case bar: return 2;
```

etc. so we probably want the current behavior of keeping any original spaces.

**Comment by yamafaktory** [comment]
What I got working is transforming:

```js
switch (foo) {
  case "foo":

  case "baz":
    doThing();
    break;
  case "bar":
    doThing();
    break;




  case "baz":
    doThing();
  case "qux":
    doThing();

  default:
    doThing();



}
```
 Into

```js
switch (foo) {
  case "foo":
  case "baz":
    doThing();
    break;

  case "bar":
    doThing();
    break;

  case "baz":
    doThing();

  case "qux":
    doThing();

  default:
    doThing();
}
```

But I still need to figuring out how to deal with the trailing comments that are break one line down in the tests (everything else is ok).

**Comment by yamafaktory** [comment]
Of course, this is very opinionated (cases grouping, one empty line after breaks, etc).

**Comment by jlongster** [comment]
I'm not entirely sure about forcing a blank line between the case statements... for some small switches I can see people wanting it to be more condensed. Does anyone else have thoughts on it?

**Comment by vjeux** [comment]
We shipped prettier 1.0 and are not going to change this anymore. The rule is that we respect blank lines when people add them all over prettier, so this is consistent. Sorry! If you really want this, your best option is to fork prettier.
